File: packages/dart/lib/src/word/rand_time.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// When something happens, from a moment to a season.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.time] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randTime(language: WordLanguage.ko, count: 3); // [새벽, 한여름, 찰나]
/// randTime(language: WordLanguage.en, count: 3); // [Twilight, Solstice, Eternity]
/// ```
List<String> randTime({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.time,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);

File: packages/javascript/lib/word/randTime.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * When something happens, from a moment to a season.
 *
 * `randWord({ theme: 'time' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randTime({ language: 'ko', count: 3 }); // ['새벽', '한여름', '찰나']
 * randTime({ language: 'en', count: 3 }); // ['Twilight', 'Solstice', 'Eternity']
 */
export function randTime(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Time words with the language and theme each one came from.
 *
 * @example
 * randTime({ language: 'ko', output: 'detail' });
 * // [{ word: '새벽', language: 'ko', theme: 'time' }]
 */
export function randTime(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randTime(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('time', options);
}
